contract_comm/evm.go

```diff
@@ -57,12 +58,12 @@ func MakeStaticCallWithAddress(scAddress common.Address, abi abi.ABI, funcName s
 	return makeCallFromSystem(scAddress, abi, funcName, args, returnObj, gas, nil, header, state, true)
 }
 
-func GetRegisteredAddress(registryId [32]byte, header *types.Header, state vm.StateDB) (*common.Address, error) {
+func GetRegisteredAddress(registryId [32]byte, header *types.Header, state vm.StateDB) (common.Address, error) {
```

Comment:
Why the switch from pointer to non pointer return value here? I think the pointer approach is actually superior to the non pointer approach because a function that tries to use a nil pointer without checking it will panic, whereas a function that tries use a zero address without checking it, may succeed (E.G a send to a zero address will have no problem, the user might take issue with it though).

Reply:
I disagree... If you're a caller and you disregard the `err` return value.. then good luck! You're already violating the contract. Since the contract of any golang function clearly states that if `err!=nil` disregard the value.

Also, if you see @oneeman comment https://github.com/celo-org/celo-blockchain/pull/1465#discussion_r605266455. We're actually making the mistake of using pointer values for comparisons, thus only comparing if they reference to the same variable, not if they are the same value. I worry much more about that kind of error, that of someone disregarding the err return value

Reply:
> Since the contract of any golang function clearly states that if err!=nil disregard the value.

Is this explicitly stated somewhere? `io.Reader` (https://golang.org/pkg/io/#Reader) violates this, and so I'm wondering how established this convention is.

I ask out of curiosity, not suggesting we switch back to using a pointer.

Reply:
I've checked on https://golang.org/doc/effective_go#functions but didn't find anything. 

I'd argue is only natural to assume that at least stated otherwise. If a function returns an error, it can happen at several moment on the code flow, does unless the function returns a default value or a "processed up to this point" value, then it seems hard to make any assumption on the value when `err != nil`